niworkflows/utils/spaces.py

```diff
@@ -0,0 +1,519 @@
+         Space(name='MNIPediatricAsym', spec={'cohort': '6', 'res': '2'})]
+
+        >>> Space.from_string("MNIPediatricAsym:cohort-5:cohort-6:res-2:res-iso1.6mm")
+        [Space(name='MNIPediatricAsym', spec={'cohort': '5', 'res': '2'}),
+         Space(name='MNIPediatricAsym', spec={'cohort': '5', 'res': 'iso1.6mm'}),
+         Space(name='MNIPediatricAsym', spec={'cohort': '6', 'res': '2'}),
+         Space(name='MNIPediatricAsym', spec={'cohort': '6', 'res': 'iso1.6mm'})]
+
+        """
+        _args = value.split(':')
+        spec = defaultdict(list, {})
+        for modifier in _args[1:]:
+            mitems = modifier.split('-', 1)
+            spec[mitems[0]].append(len(mitems) == 1 or mitems[1])
+
+        allspecs = _expand_entities(spec)
+
+        return [cls(_args[0], s) for s in allspecs]
+
+
+class SpatialReferences:
+    """
+    Manage specifications of spatial references.
+
+    Examples
+    --------
+    >>> sp = SpatialReferences([
+    ...     'func',
+    ...     'fsnative',
+    ...     'MNI152NLin2009cAsym',
+    ...     'anat',
+    ...     'fsaverage5',
+    ...     'fsaverage6',
+    ...     ('MNIPediatricAsym', {'cohort': '2'}),
+    ...     ('MNI152NLin2009cAsym', {'res': 2}),
+    ...     ('MNI152NLin2009cAsym', {'res': 1}),
+    ... ])
+    >>> sp.get_nonstd_spaces()
+    ['func', 'fsnative', 'anat']
+
+    >>> sp.get_nonstd_spaces(dim=(3,))
+    ['func', 'anat']
+
+    >>> sp.get_nonstd_spaces(only_names=False, dim=(3,))
+    [Space(name='func', spec={}),
+     Space(name='anat', spec={})]
+
+    >>> sp.get_std_spaces()
+    ['MNI152NLin2009cAsym', 'fsaverage', 'MNIPediatricAsym:cohort-2']
+
+    >>> sp.get_std_spaces(dim=(3,))
+    ['MNI152NLin2009cAsym', 'MNIPediatricAsym:cohort-2']
+
+    >>> sp.get_fs_spaces()
+    ['fsnative', 'fsaverage5', 'fsaverage6']
+
+    >>> sp.get_templates()
+    [Space(name='fsaverage', spec={'den': '10k'}),
+     Space(name='fsaverage', spec={'den': '41k'}),
+     Space(name='MNI152NLin2009cAsym', spec={'res': 2}),
+     Space(name='MNI152NLin2009cAsym', spec={'res': 1})]
+
+    >>> sp.add(('MNIPediatricAsym', {'cohort': '2'}))
+    >>> sp.get_std_spaces(dim=(3,))
+    ['MNI152NLin2009cAsym', 'MNIPediatricAsym:cohort-2']
+
+    >>> sp += [('MNIPediatricAsym', {'cohort': '2'})]
+    Traceback (most recent call last):
+      ...
+    ValueError: space ...
+
+    >>> sp += [('MNIPediatricAsym', {'cohort': '1'})]
+    >>> sp.get_std_spaces(dim=(3,))
+    ['MNI152NLin2009cAsym', 'MNIPediatricAsym:cohort-2', 'MNIPediatricAsym:cohort-1']
+
+    >>> sp.insert(0, ('MNIPediatricAsym', {'cohort': '3'}))
+    >>> sp.get_std_spaces(dim=(3,))
+    ['MNIPediatricAsym:cohort-3',
+     'MNI152NLin2009cAsym',
+     'MNIPediatricAsym:cohort-2',
+     'MNIPediatricAsym:cohort-1']
+
+    >>> sp.insert(0, ('MNIPediatricAsym', {'cohort': '3'}))
+    Traceback (most recent call last):
+      ...
+    ValueError: space ...
+
+    """
+
+    __slots__ = ('_spaces')
```

Comment:
This is currently interpreted as `{'_', 's', 'p', 'a', 'c', 'e'}`. See https://1104-74174878-gh.circle-artifacts.com/0/root/project/docs/_build/no_version_html/html/api/niworkflows.utils.spaces.html#niworkflows.utils.spaces.SpatialReferences.a

```suggestion
    __slots__ = ('_spaces',)
```

Reply:
good catch